### ISSUE TITLE
IOS - No OpKernel was registered to support Op 'Add' with these attrs

### ISSUE THREAD

**Issue by gorBaghdasaryan** [tensorflow/tensorflow#3358]
Hi everyone,

We are trying to load a very simple graph inside IOS 
Right now, we have this following error:

```
Running model failed: Invalid argument: No OpKernel was registered to support Op 'Add' with these attrs
     [[Node: deconv2d_1_1/add_3 = Add[T=DT_INT32](deconv2d_1_1/Squeeze_2, deconv2d_1_1/add_3/y)]]
```

**Comment by gorBaghdasaryan** [comment]
a code in which an error is thrown

```python
def deconvolution_2d(x, in_channels, out_channels, ksize, stride=1, pad=0, name=None):
        with tf.variable_scope(name) as scope:
            kernel = FastStyleNet.make_var('weights', [ksize, ksize, out_channels, in_channels])

            def get_deconv_outsize(size, k, s, p, cover_all=False):
                res = s * (size - 1) + k - 2 * p
                return res

            x_s = tf.shape(x)
            tttt = x_s[0]
            bbbb = get_deconv_outsize(x_s, ksize, stride, pad)
            output_shape = tf.pack(
                [tttt, bbbb[1], bbbb[2],
                 out_channels])

            return tf.nn.conv2d_transpose(x, kernel, output_shape, [1, stride, stride, 1])
```

**Comment by gorBaghdasaryan** [comment]
@petewarden  Could you take a look at this please?

**Comment by petewarden** [comment]
Are you still hitting this error @gorBaghdasaryan ? Can you send me a link to a graphdef .pb file that I can try loading if so?

**Comment by drpngx** [comment]
Closing due to inactivity. Feel free to open a new github issue if the problem still persists in recent versions.

**Comment by fegorodscy** [comment]
Hi, I have the same issue.

No OpKernel was registered to support Op 'Add' with these attrs.  Registered devices: [CPU], Registered kernels:
  device='CPU'; T in [DT_FLOAT]

	 [[Node: resnet_v1_101/block4/unit_1/bottleneck_v1/conv2/convolution/required_space_to_batch_paddings/add = Add[T=DT_INT32](resnet_v1_101/block4/unit_1/bottleneck_v1/conv2/convolution/Gather, resnet_v1_101/block4/unit_1/bottleneck_v1/conv2/convolution/required_space_to_batch_paddings/strided_slice)]]

I froze the graph using the python tool, and when I create a session loaging this graph in c++ it creashes with this error

**Comment by gepolv** [comment]
I have the same error. 
Tensorflow version: 1.1.0

**Comment by StarRain-L** [comment]
@gepolv Did you solve the problem?  I met a similar error.

> No OpKernel was registered to support Op 'Maximum' with these attrs.  Registered devices: [CPU], Registered kernels:
  device='CPU'; T in [DT_FLOAT]

	 [[Node: gradients/Mean_grad/Maximum = Maximum[T=DT_INT32, _output_shapes=[[]], _device="/device:CPU:0"](gradients/Mean_grad/Prod_1, gradients/Mean_grad/Maximum/y)]]

>
